src/Http/Authentication.Abstractions/src/TokenExtensions.cs

```diff
@@ -36,7 +36,7 @@ public static void StoreTokens(this AuthenticationProperties properties, IEnumer
         {
             if (token.Name is null)
             {
-                throw new ArgumentNullException(nameof(tokens), "Token name cannot be null.");
+                throw new ArgumentException("Token name cannot be null for any token.", nameof(tokens));
```

Comment:
This feels like an incorrect change since `token.Name` _is_ null here. It seems like what we want to do here is note that it is for `nameof(AuthenticationToken.Name)`?